homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -721,6 +723,76 @@ class EnvoyEnchargeAggregateSensorEntityDescription(SensorEntityDescription):
 )
 
 
+@dataclass(frozen=True, kw_only=True)
+class EnvoyAcbBatterySensorEntityDescription(SensorEntityDescription):
+    """Describes an Envoy ACB Battery sensor entity."""
+
+    value_fn: Callable[[EnvoyACBPower], int | str]
+
+
+ACB_BATTERY_POWER_SENSORS = (
+    EnvoyAcbBatterySensorEntityDescription(
+        key="acb_power",
+        native_unit_of_measurement=UnitOfPower.WATT,
+        device_class=SensorDeviceClass.POWER,
+        value_fn=attrgetter("power"),
+    ),
+    EnvoyAcbBatterySensorEntityDescription(
+        key="acb_soc",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        value_fn=attrgetter("state_of_charge"),
+    ),
+    EnvoyAcbBatterySensorEntityDescription(
+        key="acb_battery_state",
+        translation_key="acb_battery_state",
+        value_fn=attrgetter("state"),
```

Comment:
What are the possible states?

Reply:
As we know it now, its [discharging/idle/charging](https://pyenphase.readthedocs.io/en/latest/model_autodoc.html#pyenphase.models.acb.EnvoyACBPower.state). But maybe more, therefor left it as a string entity.

Reply:
Should we make it an enum sensor from the start and see what happens?

Reply:
Just found Full is also possible. It's a read only field anyway. If enum is more efficient for store and display we can try.

Reply:
I mean an enum device class btw. It's not more efficient, but it will provide users with both translations and the ability to select the state in the automation editor

Reply:
Ok, makes sense, I'll have a look at it

Reply:
Just realized that needs to be done in the pyenphase library. So that will be multi-stage review activity with a dependency bump. Can we leave it as is now and change to enum class in a next version?

Reply:
It does not have to be an enum enum, you can just set `options=["......"]` in the entity description, set the device class and add the translations

Reply:
Ah, thanks for the lead. Found an example.